qiskit/providers/ibmq/credentials/credentials.py

```diff
@@ -119,6 +123,26 @@ def connection_parameters(self) -> Dict[str, Any]:
 
         return request_kwargs
 
+    def _prepend_protocol_if_needed(self) -> None:
+        """Prepend the proxy URLs with protocol if needed."""
+        if 'urls' not in self.proxies:
+            return
+
+        warning_issued = False
+        for key, url in self.proxies['urls'].items():
+            colon_pos = url.find(':')
```

Comment:
Non blocking, as I did not dig deep enough into the block - but would some of the utils in https://docs.python.org/3/library/urllib.parse.html help making the logic a bit more robust? Even if it seems they are part of the upstream issue, on 3.7.5 it seems possible to provide a default `scheme` gracefully-ish:

```python
In [20]: urlparse('1.2.3.4')                                                                                                        
Out[20]: ParseResult(scheme='', netloc='', path='1.2.3.4', params='', query='', fragment='')

In [21]: urlparse('1.2.3.4', scheme='http')                                                                                         
Out[21]: ParseResult(scheme='http', netloc='', path='1.2.3.4', params='', query='', fragment='')

In [22]: urlparse('lol://1.2.3.4', scheme='http')                                                                                   
Out[22]: ParseResult(scheme='lol', netloc='1.2.3.4', path='', params='', query='', fragment='')
```

Reply:
`urlparse()` on the problematic versions might return the IP address as scheme:
```
>>> urlparse('1.2.3.4:80', 'http')
ParseResult(scheme='1.2.3.4', netloc='', path='80', params='', query='', fragment='')
```
I suppose I can check if the returned scheme is one of `http`, `https`, or `socks5`, since they're the only supported protocols, even though digits and dot are technically valid scheme characters.

Reply:
On a second thought, I'll just stick to the original solution of finding a colon. While extremely unlikely, someone could in theory have `socks5` as their proxy hostname.